# HG changeset patch
# User David Bateman
# Date 1220382078 14400
#      Tue Sep 02 15:01:18 2008 -0400
# Node ID 9f10a7b28ae3d1785e1b4d7a65344c0f20456a1e
# Parent  082fa78595740eb7f3b7ab418ff8221d38750597
__add_datasource__.m: fix indexing of varargin

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2008-09-02  David Bateman  <dbateman@free.fr>
 
+	* plot/__add_datasource__.m: Fix indexing of varargin.
+
 	* plot/__line__.m: Remove empty clause fof if/then/else test
 	* plot/plot3.m: Correctly plot matrices.
 
 2008-08-31  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* plot/__plt2mm__.m, plot/__plt2mv__.m, plot/__plt2ss__.m,
 	plot/__plt2sv__.m, plot/__plt2vm__.m, plot/__plt2vs__.m,
 	plot/__plt2vv__.m: Do not call "set" with empty arguments.
diff --git a/scripts/plot/__add_datasource__.m b/scripts/plot/__add_datasource__.m
--- a/scripts/plot/__add_datasource__.m
+++ b/scripts/plot/__add_datasource__.m
@@ -27,25 +27,26 @@ function newargs = __add_datasource__ (f
   if (ischar (data))
     data = {data};
   endif
 
   for i = 1 : numel (data)
     addproperty (strcat (data{i}, "datasource"), h, "string", "");
   endfor
 
-  i = 1;
+  i = 0;
   newargs = {};
   while (i < numel (varargin))
-    arg = varargin{i++};
-    if (ischar (arg) && length (arg > 1) && strcmpi (arg(2:end), "datasource"))
+    arg = varargin{++i};
+    if (i != numel(varargin) && ischar (arg) &&
+    length (arg > 1) && strcmpi (arg(2:end), "datasource"))
       arg = tolower (arg);
-      val = varargin{i++};
-      if (ischar (val)) 
-	set (h, arg, val);
+      val = varargin{++i};
+      if (ischar (val))
+    set (h, arg, val);
       else
-	error ("%s: expecting data source to be a string", fcn);
+    error ("%s: expecting data source to be a string", fcn);
       endif
     else
       newargs{end + 1} = arg;
     endif
   endwhile
 endfunction
